util/codec/codec.go

```diff
@@ -482,9 +482,10 @@ func peekUvarint(b []byte) (int, error) {
 }
 
 // DecodeOneToChunk decodes one value to chunk and returns the remained bytes.
-func DecodeOneToChunk(b []byte, chk *chunk.Chunk, colIdx int, ft *types.FieldType, loc *time.Location) (remain []byte, err error) {
+// bufferedBytes is only used for DecodeBytes to avoid the cost of makeslice.
+func DecodeOneToChunk(b []byte, bufferedBytes []byte, chk *chunk.Chunk, colIdx int, ft *types.FieldType, loc *time.Location) (remain []byte, _ []byte, err error) {
```

Comment:
There are too many arguments and returned values.
How about we define a type `Decoder` that contains the `buffer`, `chunk`, and `loc`?